.github/workflows/check_format_code.yml

```diff
@@ -0,0 +1,32 @@
+name: Style Check
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
I think the matrix setup is overkill. I think we generally want all our code to be checked independant on path.

Reply:
Is there a way for us to specify "check _all_ code" with this GitHub Action? The [examples of this Action](https://github.com/jidicula/clang-format-action) only show us how to give an explicit list of paths (using matrix), not "all paths".

If it's not obvious how to make the extension do all paths, I prefer to just enumerate them exactly rather than find a completely different solution.